examples/speaker_diarization/speaker_diarization/seed_test_run.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from argparse import ArgumentParser
+from argparse import Namespace
+from typing import Callable
+from typing import Dict
+
+import pandas as pd
+from evaluator import evaluate_audio_recognition
+from workflow import Inference
+from workflow import Model
+from workflow import TestSample
+from workflow import TestSuite
+from utils import realign_labels
+
+import kolena
+from kolena.workflow.annotation import LabeledTimeSegment
+from kolena.workflow.test_run import test
+
+BUCKET = "kolena-public-datasets"
+DATASET = "ICSI-corpus"
+
+TEST_SUITE_NAMES = [
+    f"{DATASET} :: average amplitude",
+    f"{DATASET} :: zero crossing rate",
+    f"{DATASET} :: energy",
+]
+
+MODEL_A = {
+    "model family": "GCP Speech To Text",
+    "model name": "gcp-stt-video",
+    "description": "Speaker diarization model optimized for videos.",
+}
+
+MODEL_METADATA: Dict[str, Dict[str, str]] = {
+    "gcp-stt-video": MODEL_A,
+}
+
+
+def seed_test_run(
+    mod: str,
+    test_suite: TestSuite,
+    infs: pd.DataFrame,
+    align_speakers: bool
+) -> None:
+    def get_stored_inferences(
+        df: pd.DataFrame,
+    ) -> Callable[[TestSample], Inference]:
```

Comment:
Is this extra indirection necessary? Can we just declare `infer` directly and capture `df` in the scope?

Reply:
That's a good idea - get_stored_inferences has been removed in the latest commit.